app/project_type/git.py

```diff
@@ -154,7 +154,7 @@ def _fetch_project(self):
         self._git_remote_command_executor.execute(fetch_command, cwd=self._repo_directory)
 
         commit_hash = self._hash or 'FETCH_HEAD'
-        reset_command = 'git reset --hard {}'.format(commit_hash)
+        reset_command = 'git reset --hard {} --'.format(commit_hash)
```

Comment:
In a comment above this line, could you post a link to some documentation or stackoverflow page that indicates what the trailing '--' does? I can't find it on the official git docs.